qualtran/bloqs/reflection_using_prepare.py

```diff
@@ -141,7 +150,9 @@ def controlled(
             and len(control_values) == 1
             and self.control_val is None
         ):
-            return ReflectionUsingPrepare(self.prepare_gate, control_val=control_values[0])
+            return ReflectionUsingPrepare(
+                self.prepare_gate, self.extra_registers, control_val=control_values[0]
+            )
```

Comment:
Change this to `return attrs.evolve(self, control_val=control_values[0])` 
```suggestion
            return attrs.evolve(self, control_val=control_values[0])
```